### PR TITLE
Disable flaky `shouldRunAsyncSchedulingEvenIfProcessingIsBlocked` test

### LINKED ISSUE #11847: `StreamProcessorTest#shouldRunAsyncSchedulingEvenIfProcessingIsBlocked` is unstable
**Summary**

> Try to answer the following as best as possible

- How often does the test fail? Not sure, possible quite often.
- Does it block your work? Sort of.
- Do we suspect that it is a real failure? Probably not.

**Failures**

Test does not finish and hangs indefinitely: https://github.com/camunda/zeebe/actions/runs/4290924604/jobs/7475475454

**Hypotheses**

Not sure. At the very least the test should not block forever and it should time out instead.

**Details**

<details><summary>Thread dump</summary>
<pre>
"-zb-actors-10" #242 daemon prio=5 os_prio=0 cpu=4.75ms elapsed=1467.52s tid=0x00007f0d8c183f50 nid=0x4bcd waiting on condition  [0x00007f0d839f7000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@17.0.6/Native Method)
	- parking to wait for  <0x0000000511882c88> (a java.util.concurrent.CountDownLatch$Sync)
	at java.util.concurrent.locks.LockSupport.park(java.base@17.0.6/LockSupport.java:211)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(java.base@17.0.6/AbstractQueuedSynchronizer.java:715)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(java.base@17.0.6/AbstractQueuedSynchronizer.java:1047)
	at java.util.concurrent.CountDownLatch.await(java.base@17.0.6/CountDownLatch.java:230)
	at io.camunda.zeebe.stream.impl.StreamProcessorTest.lambda$shouldRunAsyncSchedulingEvenIfProcessingIsBlocked$13(StreamProcessorTest.java:500)
	at io.camunda.zeebe.stream.impl.StreamProcessorTest$$Lambda$783/0x00000008010cef50.answer(Unknown Source)
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:42)
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:103)
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29)
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:34)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:82)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:56)
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:161)
	at io.camunda.zeebe.stream.api.RecordProcessor$MockitoMock$srUaQ1Mo.process(Unknown Source)
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.batchProcessing(ProcessingStateMachine.java:340)
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.lambda$processCommand$2(ProcessingStateMachine.java:263)
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine$$Lambda$697/0x00000008010ab998.run(Unknown Source)
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:84)
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.processCommand(ProcessingStateMachine.java:263)
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.tryToReadNextRecord(ProcessingStateMachine.java:222)
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.readNextRecord(ProcessingStateMachine.java:198)
	at io.camunda.zeebe.stream.impl.StreamProcessor$$Lambda$695/0x00000008010a39c8.run(Unknown Source)
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:92)
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:45)
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:119)
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106)
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87)
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198)

"ForkJoinPool-2-worker-1" #22 daemon prio=5 os_prio=0 cpu=2114.69ms elapsed=1471.56s tid=0x00007f0e6129bc70 nid=0x4012 waiting on condition  [0x00007f0e10df8000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@17.0.6/Native Method)
	- parking to wait for  <0x0000000511890988> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(java.base@17.0.6/LockSupport.java:252)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(java.base@17.0.6/AbstractQueuedSynchronizer.java:1672)
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.get(CompletableActorFuture.java:134)
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.get(CompletableActorFuture.java:109)
	at io.camunda.zeebe.scheduler.FutureUtil.join(FutureUtil.java:21)
	at io.camunda.zeebe.scheduler.future.CompletableActorFuture.join(CompletableActorFuture.java:198)
	at io.camunda.zeebe.stream.impl.StreamPlatform$ProcessorContext.close(StreamPlatform.java:364)
	at org.agrona.CloseHelper.quietCloseAll(CloseHelper.java:64)
	at io.camunda.zeebe.stream.impl.StreamPlatformExtension$StreamProcessorTestContext.close(StreamPlatformExtension.java:122)
	at org.junit.jupiter.engine.execution.ExtensionValuesStore$$Lambda$731/0x0000000800ed4448.execute(Unknown Source)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.execution.ExtensionValuesStore.lambda$closeAllStoredCloseableValues$3(ExtensionValuesStore.java:68)
	at org.junit.jupiter.engine.execution.ExtensionValuesStore$$Lambda$454/0x000000080100e108.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(java.base@17.0.6/ForEachOps.java:183)
	at java.util.stream.ReferencePipeline$3$1.accept(java.base@17.0.6/ReferencePipeline.java:197)
	at java.util.stream.SortedOps$RefSortingSink$$Lambda$455/0x0000000800e4d400.accept(java.base@17.0.6/Unknown Source)
	at java.util.ArrayList.forEach(java.base@17.0.6/ArrayList.java:1511)
	at java.util.stream.SortedOps$RefSortingSink.end(java.base@17.0.6/SortedOps.java:395)
	at java.util.stream.Sink$ChainedReference.end(java.base@17.0.6/Sink.java:258)
	at java.util.stream.AbstractPipeline.copyInto(java.base@17.0.6/AbstractPipeline.java:510)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(java.base@17.0.6/AbstractPipeline.java:499)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(java.base@17.0.6/ForEachOps.java:150)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(java.base@17.0.6/ForEachOps.java:173)
	at java.util.stream.AbstractPipeline.evaluate(java.base@17.0.6/AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(java.base@17.0.6/ReferencePipeline.java:596)
	at org.junit.jupiter.engine.execution.ExtensionValuesStore.closeAllStoredCloseableValues(ExtensionValuesStore.java:68)
	at org.junit.jupiter.engine.descriptor.AbstractExtensionContext.close(AbstractExtensionContext.java:80)
	at org.junit.jupiter.engine.execution.JupiterEngineExecutionContext.close(JupiterEngineExecutionContext.java:53)
	at org.junit.jupiter.engine.descriptor.JupiterTestDescriptor.cleanUp(JupiterTestDescriptor.java:222)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$cleanUp$1(TestMethodTestDescriptor.java:155)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor$$Lambda$730/0x0000000800ed4220.execute(Unknown Source)
</pre>
</details>

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Disable flaky test, which blocks current builds. Should be enabled when https://github.com/camunda/zeebe/issues/11849 is fixed

## Related issues

closes #11847 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by lenaschoenburg — APPROVED** [review]
Thanks! I'm starting to lose track but I think this needs to be backported too? :see_no_evil:

**Comment by Zelldon** [comment]
Good point @oleschoenburg

File: stream-platform/src/test/java/io/camunda/zeebe/stream/impl/StreamProcessorTest.java [hunk 1]
```diff
@@ -466,6 +467,7 @@ public void shouldProcessEvenIfAsyncSchedulingBlocks() throws InterruptedExcepti
   }
 
   @Test
+  @Ignore("Should be enabled when https://github.com/camunda/zeebe/issues/11849 is fixed")
```
Comment (oleschoenburg) [inline stream-platform/src/test/java/io/camunda/zeebe/stream/impl/StreamProcessorTest.java]: :x: that's a junit4 extension, I think it's called `Disabled` in JUnit 5

Comment (Zelldon) [reply stream-platform/src/test/java/io/camunda/zeebe/stream/impl/StreamProcessorTest.java]: Yeah I changed that here [4a18277](https://github.com/camunda/zeebe/pull/11851/commits/4a182772199a84d22a7a6f4f1ba3a16bc3eb1deb)

But somehow intellij is still running the test :cry:

Comment (oleschoenburg) [reply stream-platform/src/test/java/io/camunda/zeebe/stream/impl/StreamProcessorTest.java]: Sometimes IntelliJ is confused and caches run configurations. Maybe try running tests in the parent package, that should show you that this test is not run anymore.

**Pull request merged**